sycl/doc/extensions/experimental/sycl_ext_oneapi_kernel_compiler.asciidoc

```diff
@@ -172,22 +172,33 @@ a!
 ----
 namespace sycl::ext::oneapi::experimental {
 
-kernel_bundle<bundle_state::ext_oneapi_source> create_kernel_bundle_from_source(
+kernel_bundle<bundle_state::ext_oneapi_source> create_kernel_bundle_from_source(  (1)
   const context& ctxt,
   source_language lang,
   const std::string& source)
 
+kernel_bundle<bundle_state::ext_oneapi_source> create_kernel_bundle_from_source(  (2)
+  const context& ctxt,
+  source_language lang,
+  const std::vector<std::byte>& source)
+
 } // namespace sycl::ext::oneapi::experimental
 ----
 !====
 
-_Effects:_ Creates a new kernel bundle that represents a kernel that is defined
-by the source code string `source` using the language `lang`.
+_Effects (1):_ Creates a new kernel bundle that represents a kernel that is
+defined by the source code string `source` using the language `lang`.
 The bundle is associated with the context `ctxt`, and kernels from this bundle
 may only be submitted to a queue that shares the same context.
 The bundle's set of associated devices is the set of devices contained in
 `ctxt`.
 
+_Effects (2):_ Same as (1) except the source code is defined as a vector of
+bytes.
+This overload is provided for the benefit of source languages that are more
+naturally expressed as a series of raw bytes.
+However, both overloads can be used for any supported language.
```

Comment:
What would this actually mean if somebody were to pass in a SYCL source program (for example) as a series of raw bytes?  Would it be the raw bytes representing a string?  Would there be any restrictions on the encoding in that case? (e.g., would it support ASCII and UTF-8, or is it up to the implementation to somehow communicate what the bytes must be?)

Reply:
I was waffling on this decision, but I think you're right.  I don't want to specify the character encoding when reading a textual language from a vector of bytes.  Instead, I classify each source language as either text format or binary format and make it the application's responsibility to call the right overload.  See 63c8460db1acb9de3d8cfd7b55eb174cf5f7591c.

Reply:
I think this is a good solution.  If somebody really wants to store an OpenCL/SYCL string in byte form, it's not that hard for them to load it into a `std::string` before compilation.